File: fileio/game_modes.go
package fileio

// Game mode constants
const (
	GameModeCampaign   = 1  // Campaign mode
	GameModeConquest   = 2  // Conquest mode
	GameModeFrontier   = 6  // Frontier mode
	GameModeArmyGroup  = 10 // Army group mode (armygroup.sav)
)

// GetGameModeName returns a human-readable name for the game mode
func GetGameModeName(gameMode uint32) string {
	switch gameMode {
	case GameModeCampaign:
		return "Campaign"
	case GameModeConquest:
		return "Conquest"
	case GameModeFrontier:
		return "Frontier"
	case GameModeArmyGroup:
		return "ArmyGroup"
	default:
		return "Unknown"
	}
}

File: src/WC4SaveEditor.Core/Data/GameModes.cs
namespace WC4SaveEditor.Core.Data;

public static class GameModes
{
    public const uint Campaign = 1;
    public const uint Conquest = 2;
    public const uint Frontier = 6;
    public const uint ArmyGroup = 10;

    public static string GetName(uint gameMode) => gameMode switch
    {
        Campaign => "Campaign",
        Conquest => "Conquest",
        Frontier => "Frontier",
        ArmyGroup => "ArmyGroup",
        _ => "Unknown"
    };
}
